### ISSUE TITLE
weave not in scipy 0.17

### ISSUE THREAD

**Issue by mathause** [jakobrunge/tigramite_old#1]
How about (in `tigramite_estimation_beta.py` ):

```
#  Import scipy.weave for C++ inline code and other packages
from scipy import linalg, special, stats, spatial

try:
    from scipy import weave
except ImportError:
    import weave
```

**Comment by jakobrunge** [comment]
I added the line, but actually the scipy docs still mention weave: http://docs.scipy.org/doc/scipy/reference/ 
I use scipy version 0.16 from the Anaconda package and it works...

**Comment by mathause** [comment]
Yes, that is true - maybe I didn't compile it. But they also mention that they will deprecate it in the future and recommend to use the 'standalone' `weave` package. The way you changed the code I just run into the next import error (i.e. you really need to use `import weave`).

**Comment by jakobrunge** [comment]
Indeed weave is not part of scipy 0.17 anymore. I have now in the latest commit replaced all weave code by Cython code (in "tigramite_cython_code.pyx"). I included the compiled *.so files and also the setup file which can by run with

python setup_tigramite_cython_code.py build_ext --inplace

to generate the compiled tigramite_cython_code.so for other machines.

Another thing: Scipy 0.17 fixes a memory leak in the cKDTree package which I use as part of the 'cmi_knn' estimator. I recommend an update to scipy 0.17, eg using anaconda!
